File: test/fixtures/java-unirest/post_with_colon_in_header.java
import kong.unirest.HttpResponse;
import kong.unirest.Unirest;

class Main {
    public static void main(String[] args) {
            HttpResponse<String> httpResponse
               = Unirest.post("http://localhost:28139/endpoint")
            .header("Content-Type", "application/json")
            .header("key", "abcdefg")
            .asString();
    }
}

File: test/fixtures/kotlin-ktor/post_with_colon_in_header.kt
import io.ktor.client.HttpClient
import io.ktor.client.request.header
import io.ktor.client.request.post
import io.ktor.client.statement.bodyAsText
import io.ktor.http.isSuccess
import kotlinx.coroutines.runBlocking

val client = HttpClient()

runBlocking {
    val response = client.post("http://localhost:28139/endpoint") {
        header("Content-Type", "application/json")
        header("key", "abcdefg")
    }
    if (!response.status.isSuccess()) throw Exception("Request failed with status code ${response.status}")

    println(response.bodyAsText())
}
